libraries/AP_Volz_Protocol/AP_Volz_Protocol.cpp

```diff
@@ -0,0 +1,166 @@
+    // @DisplayName: Enable of volz servo protocol
+    // @Description: outputs volz servo protocol to serial out.
+    // @Values: 0:Disabled,1:Enabled
+    // @User: Standard
+    AP_GROUPINFO("ENABLED", 0, AP_Volz_Protocol, enabled, 0),
+
+    // @Param: CHAN_BITMASK
+    // @DisplayName: Channel Bitmask
+    // @Description: Enable of volz servo protocol to specific channels
+	// @Values: 0,1....
+    // @User: Standard
+    AP_GROUPINFO("BITMASK", 1, AP_Volz_Protocol, chan_bitmask,  0),
+
+    // @Param: MIN_PWM
+    // @DisplayName: Minimum PWM Mapping
+    // @Description: Enable of volz servo protocol to specific channels
+	// @Values: 0,1....
+    // @User: Standard
+    AP_GROUPINFO("MIN_PWM", 2, AP_Volz_Protocol, min_pwm,  900),
+
+    // @Param: MAX_PWM
+    // @DisplayName: Miximum PWM Mapping
+    // @Description: Enable of volz servo protocol to specific channels
+	// @Values: 0,1....
+    // @User: Standard
+    AP_GROUPINFO("MAX_PWM", 3, AP_Volz_Protocol, max_pwm,  2100),
+
+
+    AP_GROUPEND
+};
+
+// Constructor
+AP_Volz_Protocol::AP_Volz_Protocol()
+{
+	AP_Param::setup_object_defaults(this, var_info);
+}
+
+AP_Volz_Protocol AP_Volz_Protocol::create(){
+	return AP_Volz_Protocol{};
+}
+
+void AP_Volz_Protocol::init(const AP_SerialManager& serial_manager){
+	if(enabled != 1)
+		return;
+
+	if(min_pwm < 900)
+		_min_pwm = 900;
+	else
+		_min_pwm = min_pwm;
+
+	if(max_pwm > 2100)
+		_max_pwm = 2100;
+	else
+		_max_pwm = max_pwm;
+
+	_pwm_range = (_max_pwm - _min_pwm);
+	_chan_bitmask = chan_bitmask;
+
+	if((_port = serial_manager.find_serial(AP_SerialManager::SerialProtocol_Volz,0))){
+		_initialized = true;
+	}
+}
+
+void AP_Volz_Protocol::update(){
+	gcs().send_text(MAV_SEVERITY_CRITICAL, "Volz PARAMS: %u %u %u %u", enabled, chan_bitmask, min_pwm, max_pwm);
+
+	if(_initialized != true)
+		return;
+
+	uint64_t tnow = AP_HAL::micros64();
+	uint16_t pwm_values[16];
+
+	if(!SRV_Channels::get_all_outputs_pwm(pwm_values,16)){
+		gcs().send_text(MAV_SEVERITY_CRITICAL, "Volz Protocol failed to get pwm values, don't fly.");
+		return;
+	}
+
+
+	uint8_t i,sent = 0;
+
+	// loop for all 16 channels
+	for (i=0;i<16;i++){
+		// check if current channel is needed for Volz protocol
+		if((_chan_bitmask >> i) & 0x0001){
+			uint16_t value;
+			// check if current channel PWM is within range
+			if(pwm_values[i] - _min_pwm < 0)
+				value = 0;
+			else
+				value = pwm_values[i] - _min_pwm;
+
+			// scale the PWM value to Volz value
+			value = value + VOLZ_EXTENDED_POSITION_MIN;
+			value = value * VOLZ_SCALE_VALUE / _pwm_range;
+
+			// make sure value stays in range
+			if(value > VOLZ_EXTENDED_POSITION_MAX)
+				value = VOLZ_EXTENDED_POSITION_MAX;
+
+			// prepare Volz protocol data.
+			uint8_t data[4];
+
+			data[0] = VOLZ_SET_EXTENDED_POSITION_CMD;
+			data[1] = i + 1;		// send actuator id as 1 based index so ch1 will have id 1, ch2 will have id 2 ....
+			data[2] = HIGHBYTE(value);
+			data[3] = LOWBYTE(value);
+
+			send_command(data);
```

Comment:
For clarity, would you mind changing this lines to be something like:

line 131: _port->write(data, 4);
line 132: calculate_and_send_crc(data);